src/htmx.js

```diff
@@ -1400,19 +1400,20 @@ var htmx = (function() {
    * @param {string} oobValue
    * @param {Element} oobElement
    * @param {HtmxSettleInfo} settleInfo
-   * @returns
+   * @param {SwapOptions} swapOptions
    */
-  function oobSwap(oobValue, oobElement, settleInfo) {
+  function oobSwap(oobValue, oobElement, settleInfo, swapOptions) {
+    const swapSpec = getSwapSpecification(oobElement, oobValue)
+
     let selector = '#' + getRawAttribute(oobElement, 'id')
-    /** @type HtmxSwapStyle */
-    let swapStyle = 'outerHTML'
-    if (oobValue === 'true') {
-      // do nothing
-    } else if (oobValue.indexOf(':') > 0) {
-      swapStyle = oobValue.substr(0, oobValue.indexOf(':'))
-      selector = oobValue.substr(oobValue.indexOf(':') + 1, oobValue.length)
-    } else {
-      swapStyle = oobValue
+    let swapStyle = swapSpec.swapStyle
+
+    if (swapStyle.indexOf(':') > 0) {
+      selector = swapStyle.substr(swapStyle.indexOf(':') + 1, swapStyle.length)
+      swapStyle = swapStyle.substr(0, swapStyle.indexOf(':'))
```

Comment:
Random drive by optimization / modernization:
```suggestion
    const colonIdx = swapStyle.indexOf(':')
    if (colonIdx > 0) {
      selector = swapStyle.substring(colonIdx + 1)
      swapStyle = swapStyle.substring(0, colonIdx)
```